File: Assets/Old/Scripts/Trash/Exercise.js
import System.Collections.Generic;
import System.Xml.Serialization;

@XmlRoot("EXERCISE")
public class EXERCISE {

	@XmlAttribute("initialId")
	public var initialId ;
	
	@XmlAttribute("finalId")
	public var finalId ;
	@XmlIgnoreAttribute
	public var initialArt = "";
	@XmlIgnoreAttribute
	public var finalArt = "";
	
	@XmlElement("ANGLE")
	public var ang : Angle;
	
	@XmlElement("AXIS")
	public var eje : Eje;
	

	@XmlElement("INI")
	public var ini : Ini;
	
	@xmlElement("REFERENCE")
	public var reference : Reference;
	
	
	@XmlElement("RESTRICTION")
	public var restrictions : List.<Restriction> = new List.<Restriction>();


	public function Save(path : String) {
		var ns : XmlSerializerNamespaces = new XmlSerializerNamespaces();
		ns.Add("","");
	
 		var serializer : XmlSerializer = new XmlSerializer(EXERCISE);
 		var stream : Stream = new FileStream(path, FileMode.Create);
 		serializer.Serialize(stream, this, ns);
 		stream.Close();
 	}

}

public class Reference {
	
	@XmlAttribute("id")
	public var id ;
	@XmlIgnoreAttribute
	public var nameId = "";
	
	@XmlAttribute("x")
	public var x : String = "";
	
	@XmlAttribute("y")
	public var y : String = "";
	
	@XmlAttribute("z")
	public var z : String = "";
}

public class Angle {
	@XmlAttribute("min")
	public var Min : String = "";
	
	@XmlAttribute("max")
	public var Max : String = "";
}

public class Eje {
	@XmlAttribute("x")
	public var X : String = "";
	
	@XmlAttribute("y")
	public var Y : String = "";
	
	@XmlAttribute("z")
	public var Z : String = "";
}

public class Ini {

	@XmlAttribute("x")
	public var x : String = "";
	
	@XmlAttribute("y")
	public var y : String = "";
	
	@XmlAttribute("z")
	public var z : String = "";
}

public class Restriction {

	@XmlElement("initialId")
	public var initialId : int;
    @XmlElement("finalId") 
	public var finalId : int;
	@XmlIgnoreAttribute
	public var initialArt = "";	
	@XmlIgnoreAttribute	
	public var finalArt = "";
	
	
	
	@XmlElement("x")
	public var x : int;
	@XmlElement("y")
	public var y : int;
	@XmlElement("z")
	public var z : int;

	@XmlElement("grade")
	public var grade = 0;
}

File: Assets/Old/Scripts/Interface/Exercise.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using System.Xml.Serialization;
using System.IO;

[XmlRootAttribute("EXERCISE")]
public class Exercise {

	[XmlAttribute("initialId")]
	public int initialId;

	[XmlAttribute("finalId")]
	public int finalId;

	[XmlIgnoreAttribute]
	public string initialArt = "";

	[XmlIgnoreAttribute]
	public string finalArt = "";

	public Angle ang;
	
	public Axis axis;
		
	public Ini ini;

	public RotIni rotIni;

	public RotEnd rotEnd;
	
	public Reference reference;

	public List<Restriction> Restrictions;

	public Exercise() {
		ang = new Angle();
		axis = new Axis();
		ini = new Ini();
		rotIni = new RotIni();
		rotEnd = new RotEnd();
		reference = new Reference();
		Restrictions = new List<Restriction>();
	}
	
	public void Save(string path) {
		XmlSerializerNamespaces ns = new XmlSerializerNamespaces();
		ns.Add("","");

		XmlSerializer serializer = new XmlSerializer(typeof(Exercise));
		Stream stream = new FileStream(path, FileMode.Create);
		serializer.Serialize(stream, this, ns);
		stream.Close();
	}
}

public class Reference {
	
	[XmlAttribute("id")]
	public int id;

	[XmlIgnoreAttribute]
	public string nameId = "";
	
	[XmlAttribute("x")]
	public string x = "";
	
	[XmlAttribute("y")]
	public string y = "";
	
	[XmlAttribute("z")]
	public string z = "";

	public Reference() {}
}

public class Angle {
	[XmlAttribute("min")]
	public string Min = "0";
	
	[XmlAttribute("max")]
	public string Max = "0";

	public Angle() {}
}

public class Axis {
	[XmlAttribute("x")]
	public string X = "";
	
	[XmlAttribute("y")]
	public string Y = "";
	
	[XmlAttribute("z")]
	public string Z = "";

	public Axis() {}
}

public class Ini {
	[XmlAttribute("x")]
	public string x = "";
	
	[XmlAttribute("y")]
	public string y = "";
	
	[XmlAttribute("z")]
	public string z = "";

	public Ini() {}
}

public class RotIni {
	[XmlAttribute("x")]
	public string x = "";
	
	[XmlAttribute("y")]
	public string y = "";
	
	[XmlAttribute("z")]
	public string z = "";
	
	public RotIni() {}
}

public class RotEnd {
	[XmlAttribute("x")]
	public string x = "";
	
	[XmlAttribute("y")]
	public string y = "";
	
	[XmlAttribute("z")]
	public string z = "";
	
	public RotEnd() {}
}

public class Restriction {
	[XmlElement("initialId")]
	public int initialId;

	[XmlElement("finalId")] 
	public int finalId;

	[XmlIgnoreAttribute]
	public string initialArt = "";	

	[XmlIgnoreAttribute]	
	public string finalArt = "";

	public int x, y, z;

	public int rotX, rotY, rotZ;
	
	public int grade = 0;

	public Restriction() {}
}
